src/lib.rs

```diff
@@ -437,21 +437,8 @@ impl<T> Event<T> {
         // Make sure the notification comes after whatever triggered it.
         notify.fence(notify::Internal::new());
 
-        if let Some(inner) = self.try_inner() {
-            let limit = if notify.is_additional(Internal::new()) {
-                usize::MAX
-            } else {
-                notify.count(Internal::new())
-            };
-
-            // Notify if there is at least one unnotified listener and the number of notified
-            // listeners is less than `limit`.
-            if inner.needs_notification(limit) {
-                return inner.notify(notify);
-            }
-        }
-
-        0
+        let inner = unsafe { &*self.inner() };
```

Comment:
I guess try_inner -> inner change causes UB (null-ptr-deref) on `Event::new().notify()`.

Reply:
The `inner` function initializes the inner pointer if it's null, so this shouldn't be an issue.

Reply:
Oh, sorry, you're right.